adam/learner/learner_utils.py

```diff
@@ -781,3 +790,33 @@ def get_slot_from_semantic_node(
         if object_node.concept == object_concept:
             return slot_var.name
     return slot
+
+
+def get_root_if_perception_is_object_cluster(
+    graph: DiGraph, matched_subgraph_nodes: AbstractSet[PerceptionGraphNode]
+) -> Optional[ObjectClusterNode]:
+    """Get the root object cluster node from a graph given a subset of nodes which represent the object.
+
+    If the root object cluster does not exist this function will return none.
```

Comment:
I think it's worth being explicit here about why the root object cluster might not exist, because otherwise the name and docstring don't obviously make sense together. Something like:

```suggestion
    If a root object cluster node does not exist this function will return none. For example, this happens when the passed nodes represent an action and not an object. It could also happen if the nodes represent two separate objects.
```